### PR TITLE
Support bootstrapping DDC for web workers

### PR DESCRIPTION
**Pull request by kevmoo**
Fixes https://github.com/dart-lang/build/issues/1059

### CODE REVIEW

**Comment by kevmoo** [comment]
CC @matanlurey @natebosch

File: build_web_compilers/lib/src/dev_compiler_bootstrap.dart [hunk 1]
```diff
@@ -137,20 +137,33 @@ String _entryPointJs(String bootstrapModuleName) => '''
...
-  el.src = baseUrl + "packages/\$sdk/dev_compiler/amd/require.js";
-  el.setAttribute("data-main", _currentDirectory + "$bootstrapModuleName");
-  document.head.appendChild(el);
-})();
+
+  var mapperUri = baseUrl + "packages/\$sdk/dev_compiler/web/dart_stack_trace_mapper.js";
+  var requireUri = baseUrl + "packages/\$sdk/dev_compiler/amd/require.js";
+  var mainUri = _currentDirectory + "$bootstrapModuleName";
+
+  if (typeof document != 'undefined') {
```
Comment (kevmoo) [inline build_web_compilers/lib/src/dev_compiler_bootstrap.dart]: Or we could be doing something like this - https://stackoverflow.com/a/18002694/39827

**Comment by kevmoo** [comment]
Changelog added – filed https://github.com/dart-lang/build/issues/1117

File: build_web_compilers/CHANGELOG.md [hunk 2]
```diff
@@ -1,4 +1,4 @@
-# 0.3.4
+# 0.3.4-dev
```
Comment (kevmoo) [inline build_web_compilers/CHANGELOG.md]: We've generally feel this is not needed. One less thing to fix when you publish.

...but no big deal...

Comment (jakemac53) [reply build_web_compilers/CHANGELOG.md]: hmm ok we have been adding this for the build packages at least

**Pull request merged**